src/Steam.Common/SteamId.cs

```diff
@@ -0,0 +1,492 @@
+            var universe = (Universe)int.Parse(match.Groups[2].Value);
+            var account = uint.Parse(match.Groups[3].Value);
+            uint instance = match.Groups[4].Success ? uint.Parse(match.Groups[4].Value) : 0;
+            AccountType type;
+            switch (typeChar)
+            {
+                case 'A':
+                    type = AccountType.AnonGameServer;
+                    break;
+                case 'G':
+                    type = AccountType.GameServer;
+                    break;
+                case 'C':
+                    type = AccountType.ContentServer;
+                    break;
+                case 'g':
+                    type = AccountType.Clan;
+                    instance = 0;
+                    break;
+                case 'c':
+                    type = AccountType.Chat;
+                    instance = ClanFlag;
+                    break;
+                case 'L':
+                    type = AccountType.Chat;
+                    instance = LobbyFlag;
+                    break;
+                case 'T':
+                    type = AccountType.Chat;
+                    instance = 0;
+                    break;
+                case 'U':
+                    type = AccountType.Individual;
+                    instance = 1;
+                    break;
+                case 'I':
+                case 'i':
+                    type = AccountType.Invalid;
+                    instance = 1;
+                    break;
+                default:
+                    throw new InvalidOperationException("unreachable");
+            }
+
+            return new SteamId(account, type, universe, instance);
+        }
+
+        /// <summary>
+        /// Returns whether a given SteamID is considered valid
+        /// </summary>
+        /// <param name="id">The ID to check</param>
+        public static bool IsValid(SteamId id)
+        {
+            if (id.AccountType <= AccountType.Invalid || id.AccountType > AccountType.AnonUser)
+                return false;
+
+            if (id.AccountUniverse <= Universe.Invalid || id.AccountUniverse > Universe.Dev)
+                return false;
+
+            if (id.AccountType == AccountType.Individual)
+            {
+                if (id.AccountId == 0 || id.AccountInstance > (uint)Instance.Web)
+                    return false;
+            }
+
+            if (id.AccountType == AccountType.Clan)
+            {
+                if (id.AccountId == 0 || id.AccountInstance != 0)
+                    return false;
+            }
+
+            if (id.AccountType == AccountType.GameServer)
+            {
+                if (id.AccountId == 0)
+                    return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns whether this is a login ID to be filled in
+        /// </summary>
+        public bool IsBlankAnonymousAccount => AccountId == 0 && IsAnonymousAccount && AccountInstance == 0;
```

Comment:
I'd like to see one white line between properties to make the distinction more clear.